src/core/public/chrome/ui/header/header.tsx

```diff
@@ -287,9 +299,18 @@ class HeaderUI extends Component<Props, State> {
       .map(navLink => ({
         key: navLink.id,
         label: navLink.title,
-        href: navLink.href,
+
+        // Legacy apps use href, NP apps use onClick
+        // This needs to work with both href and onClick to support "open in new tab" correctly, however EUI
+        // does not current support this.
+        // https://github.com/elastic/eui/pull/1933
+        href: legacyMode || navLink.legacy ? navLink.href : undefined,
```

Comment:
We can simplify this to only check the condition once via:

```ts
...(legacyMode || navLink.legacy
  ? { href: navLink.href }
  : { onClick: () => navigateToApp(navLink.id) }
),
```

Reply:
I plan on fixing the caveat mentioned in the comment once EUI is upgraded. That should happen before this merges, so I'm just going to leave this as-is since it will change anyways.